internal/provider/resource_integration_test.go

```diff
@@ -0,0 +1,206 @@
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/apono-io/apono-sdk-go"
+	"github.com/google/uuid"
+	"github.com/jarcoal/httpmock"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAccIntegrationResource(t *testing.T) {
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	var integrations = map[string]*apono.Integration{}
+	httpmock.RegisterResponder(http.MethodPost, "http://api.apono.dev/api/v2/integrations", func(req *http.Request) (*http.Response, error) {
+		var createReq apono.CreateIntegration
+		if err := json.NewDecoder(req.Body).Decode(&createReq); err != nil {
+			return httpmock.NewStringResponse(400, err.Error()), nil
+		}
+
+		id, err := uuid.NewUUID()
+		if err != nil {
+			return httpmock.NewStringResponse(500, err.Error()), nil
```

Comment:
don't we want to return the error in this step?

Reply:
No, we want to return the error so it wound be printed in the provider error